drivers/network/afd/afd/bind.c

```diff
@@ -51,13 +51,13 @@ NTSTATUS WarmSocketForBind( PAFD_FCB FCB, ULONG ShareType ) {
                 Status = STATUS_NO_MEMORY;
         }
 
-        if (NT_SUCCESS(Status))
+        if (NT_SUCCESS(Status) && FCB->Recv.Content < FCB->Recv.Size)
```

Comment:
In this case we drop packets that are received last,
would it make sense to drop the first packets instead?

(I have no clue about this, merely asking to understand the reasoning behind this)

Reply:
For stream sockets, new data is "paused" (from an application standpoint) until the buffer has space again. For datagram sockets, either an existing or a new datagram has to be dropped. It was really just intuition that made me drop the new one rather than an existing one, but:
* according to a quick Google search this seems to be the expected behavior
* since there's a "peek" functionality, you can't remove the oldest if it has already been looked at, which also suggests that newer packets should be dropped.
I wish we had tests for this but apparently we don't... my patch essentially generates no difference on testbot.

Reply:
Thanks for explaining.

Reply:
Writing a test for this should be relatively easy given an app that receives datagrams and waits a long time between receives.  Fairly certain this is closer to correct.

Reply:
Yup I'm actually working on one already. Just have to deal with an unrelated regression (CS-14103) first.